### ISSUE TITLE
publicOnly option for require-jsdoc not working

### ISSUE THREAD

**Issue by jatwood** [gajus/eslint-plugin-jsdoc#358]
Our project uses the publicOnly option but we've found the rule still applies to functions that are not exported from the module. We've reproduced it in the the plugin test cases, here is a link to the diff. 

https://github.com/gajus/eslint-plugin-jsdoc/compare/master...TeachersPayTeachers:add-repro-of-publiconly-issue?expand=1

**Comment by gajus** [comment]
https://github.com/gajus/eslint-plugin-jsdoc/issues/288

**Comment by jatwood** [comment]
The linked issue seems to say that `publicOnly` isn't reporting an error for certain export types but  our issue is that it is reporting errors for non-exported functions.

**Comment by Josmithr** [comment]
I just updated to 15.8.4, and I am still seeing false positives. I.e. *all* functions are being flagged as requiring docs, not just exported ones.
Repros with the following settings:
```
"jsdoc/require-jsdoc": [
	"error",
	{
		"publicOnly": true
	}
]
```

**Comment by brettz9** [comment]
I am not seeing an issue with normal functions and `publicOnly` in 15.8.4. Can you provide a test case of code to reproduce you are testing with in addition to the config you provided?

**Comment by Josmithr** [comment]
I am no longer seeing the false-positives as of 15.9.1. Thank you!!
